go/vt/mysqlproxy/mysqlproxy.go

```diff
@@ -78,7 +78,10 @@ func (mp *Proxy) Execute(ctx context.Context, session *ProxySession, sql string,
 	}
 
 	if err != nil {
-		return nil, nil, err
+		// N.B. You must return session here. Modeled after vtgate mysql plugin, the vtqueryserver
+		// plugin expects you to return a new or updated session and not drop it on the floor during
+		// an error.
+		return session, nil, err
```

Comment:
May be simpler to always `return session, result, err`?

Reply:
sure